samcli/hook_packages/terraform/hooks/prepare.py

```diff
@@ -905,14 +908,175 @@ def _generate_makefile_rule_for_lambda_resource(
+    to the build_output_path as is created by the Terraform plan output
+
+    Parameters
+    ----------
+    sam_metadata_resource: SamMetadataResource
+        A sam metadata resource; the generated recipe jpath will correspond to building the lambda resource
+        associated with this sam metadata resource
+
+    resource_address: str
+        Full address of a Terraform resource
+
+    Returns
+    -------
+    str
+       Full JPath string for a resource from planned_values to build_output_path
+    """
+    jpath_string_template = (
+        "|values|root_module{child_modules}|resources|"
+        '[?address=="{resource_address}"]|values|triggers|built_output_path'
+    )
+    child_modules_template = "|child_modules|[?address=={module_address}]"
+    module_address = sam_metadata_resource.current_module_address
+    full_module_path = ""
+    parent_modules = _get_parent_modules(module_address)
+    for module in parent_modules:
+        full_module_path += child_modules_template.format(module_address=module)
+    return jpath_string_template.format(child_modules=full_module_path, resource_address=resource_address)
+
+
+def _get_parent_modules(module_address: Optional[str]) -> List[str]:
+    """
+    Convert an a full Terraform resource address to a list of module
+    addresses from the root module to the current module
+
+    e.g. "module.level1_lambda.module.level2_lambda" as input will return
+    ["module.level1_lambda", "module.level1_lambda.module.level2_lambda"]
+
+    Parameters
+    ----------
+    module_address: str
+       Full address of the Terraform module
+
+    Returns
+    -------
+    List[str]
+       List of module addresses starting from the root module to the current module
+    """
+    if not module_address:
+        return []
+
+    # Split the address on "." then combine it back with the "module" prefix for each module name
+    modules = module_address.split(".")
+    modules = [".".join(modules[i : i + 2]) for i in range(0, len(modules), 2)]
+
+    if not modules:
+        # The format of the address was somehow different than we expected from the
+        # module.<name>.module.<child_module_name>
+        return []
+
+    # Prefix each nested module name with the previous
+    previous_module = modules[0]
+    full_path_modules = [previous_module]
+    for module in modules[1:]:
+        module = previous_module + "." + module
+        previous_module = module
+        full_path_modules.append(module)
+    return full_path_modules
+
+
+def _get_makefile_build_target(logical_id: str) -> str:
+    """
+    Formats the Makefile rule build target string as is needed by the Makefile
+
+    Parameters
+    ----------
+    logical_id: str
+       Logical ID of the resource to use for the Makefile rule target
+
+    Returns
+    -------
+    str
+        The formatted Makefile rule build target
+    """
+    return f"build-{logical_id}:\n"
```

Comment:
it is better to use `os.linesep` instead of `\n` to handle different new line representations in each OS